### ISSUE TITLE
Separate main logic into a library with C FFI interface

### ISSUE THREAD

**Issue by namazso** [namazso/SecureUxTheme#120]
This allows developers implement theming and install features into other software.
Will also require a license change on the main dll to LGPL as well as writing the new module under that license.
ThemeTool will keep being GPL as it contains contributions, and I don't want others to include code into proprietary apps.

**Comment by Spiritreader** [comment]
I am unable to get the ThemeDll project to build for any arch.

ThemeTool and SecureUxTheme itself builds fine if I follow the instructions in the readme.
![image](https://user-images.githubusercontent.com/18438899/196179712-b48c43b3-b8bb-4ed9-b598-be8601973335.png)

But whenever trying to build the library there are errors in the resources file

![image](https://user-images.githubusercontent.com/18438899/196179815-0656b75d-892a-44c9-ac3d-ea1161df89ed.png)
![image](https://user-images.githubusercontent.com/18438899/196179842-c6a8c6b2-6664-43e3-a2e8-4d719a81c9cf.png)

It's asking for:
```
IDR_SECUREUXTHEME_DLL_X86 256                     "../bin/Release/Win32/SecureUxTheme.dll"

IDR_SECUREUXTHEME_DLL_X64 256                     "../bin/Release/x64/SecureUxTheme.dll"

IDR_SECUREUXTHEME_DLL_ARM64 256                     "../bin/Release/ARM64/SecureUxTheme.dll"
```
These definitions are also in ThemeTool.rc, but therre aren't any issues there.
It also can't find the Language definition. 

I verified that these files are there for all archs and also for the Debug/Release profile.
![image](https://user-images.githubusercontent.com/18438899/196182866-a7af4125-7fb2-4b13-abc3-3cfb13bf01e4.png)

I have the C++ v143 buildtools for x86 and ARM64.

Do I have something configured wrong?

**Comment by namazso** [comment]
did you try building ThemeDll after the SecureUxTheme.dlls were built already, like with ThemeTool? 

It passes CI (at least on x86) so the problem is probably on your side if really none of them work.

Oddly enough you're also missing ThemeLib.lib on ARM64, that might also be related?

**Comment by Spiritreader** [comment]
> Oddly enough you're also missing ThemeLib.lib on ARM64, that might also be related?

Sadly that wasn't it. I built it but still can't build ThemeDll.
![image](https://user-images.githubusercontent.com/18438899/196189409-676c18c6-e591-46eb-b5f2-c7686adbafed.png)

> did you try building ThemeDll after the SecureUxTheme.dlls were built already, like with ThemeTool?

Yes, first I built all SecureUxTheme.dlls for the debug and release config, then I attempted to build ThemeDll.

The relative path that's in the resource file, does that refer to the bin directory at the project root?
![image](https://user-images.githubusercontent.com/18438899/196190701-01cfb288-7e41-4f13-94aa-c5b787ddb7c9.png)

I'm also really confused why the language definition in the resource file doesn't work, especially since I'm able to build ThemeTool with no issues.

I'm trying to build a C# wrapper for the ThemeDll library, which is why I wanted to build with an x64 platform target. But I never seem to have a ton of luck with getting C++ projects to compile.

In any case, thanks for your help! Good to know the issues is somewhere in the depths of how Visual Studio configured my c build environment.
